### PR TITLE
add ADR 11: annotations lifespan interfaces

### PR DESCRIPTION
**Pull request by arielmirra**

### CODE REVIEW

File: adrs/0011-annotations-lifespan.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Context
+Not all annotations are needed in all stages of AMF (parsing, resolution/transformation, validation, rendering/emission).
+
+## Decision
+For this reason, the decision has been made in favor of adding 3 distinct types of annotations, that all others should implement:
+
+- `ResolvableAnnotation`
+    - An annotation that must be resolved
+    - To allow deferred resolution on unordered graph parsing
```
Comment (tomsfernandez) [inline adrs/0011-annotations-lifespan.md:15]: This phrase is pretty but it isn't clear. What is "deferred resolution" ? When is graph parsing "unordered" ?

File: adrs/0011-annotations-lifespan.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- `SerializableAnnotation`
+    - An annotation that needs to be written in the JSON-LD
```
Comment (tomsfernandez) [inline adrs/0011-annotations-lifespan.md]: `written` -> `dumped`

File: adrs/0011-annotations-lifespan.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# 10. Annotation types
+Date: 2022-07-05
```
Comment (tomsfernandez) [inline adrs/0011-annotations-lifespan.md]: The date is weird, this decision was taken a long time ago but the ADR was written now. This is shady as doesn't match with the git annotations in the code.

File: adrs/0011-annotations-lifespan.md (hunk 1 above)
Comment (arielmirra) [reply adrs/0011-annotations-lifespan.md:15]: not really sure, extracted it from the javadoc written by Pedro. It's used in the RDFParser.

File: adrs/0011-annotations-lifespan.md (hunk 3 above)
Comment (arielmirra) [reply adrs/0011-annotations-lifespan.md]: changed it to the last meaningful commit in the code 👌🏽

Comment (arielmirra) [reply adrs/0011-annotations-lifespan.md]: changed it to the last meaningful commit in the code 👌🏽

Comment (arielmirra) [reply adrs/0011-annotations-lifespan.md]: changed it to the last meaningful commit in the code 👌🏽

File: adrs/0011-annotations-lifespan.md (hunk 1 above)
Comment (pope1838) [reply adrs/0011-annotations-lifespan.md:15]: We all agree that graph our graph do not ensure order, but I think that fact has no relation with the definition of a resolvable annotation.
For me, resolvable annotations means that the value of the annotation must be processed into the rdf model while parsing it.

Comment (pcolunga) [reply adrs/0011-annotations-lifespan.md:15]: The 'resolvable' name comes from the _theoretical_ resolution/linkage phase of a compiler (usually needed, depending on the complexity of the underlying model, for resolving _references_ within the model).
When parsing from RDF, order cannot be guaranteed for nodes, properties, nor types (as with jsonld but we are taking wrong assumptions there). The idea was to enable an annotation to be resolved after parsing [with some context] (parse -> resolve -> consume).

**Review by pcolunga — DISMISSED** [review]

**Review by hghianni — APPROVED** [review]

**Pull request merged**